CRM/Contact/Import/Form/MapField.php

```diff
@@ -174,12 +171,12 @@ public function buildQuickForm() {
         $id = $first = $second = NULL;
       }
       if (($first === 'a' && $second === 'b') || ($first === 'b' && $second === 'a')) {
-        $cType = $contactRelationCache[$id]["contact_type_{$second}"];
+        $cType = $contactRelationCache[$id]["contact_type_$second"];
```

Comment:
Subjective aside: I find this old expression more readable than this new expression. (*With the new one, my eyes have to re-scan a few times to confirm that `$` really is there and recognize that `second` is a variable-name.*)

Granted, Github UI and PHPStorm UI render this differently. PHPStorm's highlighter changes the color of `$second` (which serves a similar purpose of helping you recognize the variable-name). I suppose that (within PHPStorm) the `{}`s are visually redundant. Maybe that's why they nag you to remove them.

But other highlighters don't work the same way. The change makes the code less readable in other environments.

PHPStorm's nag feels counter-productive to me. It can be fixed, though:

<img width="723" alt="Screen Shot 2023-01-10 at 5 13 14 PM" src="https://user-images.githubusercontent.com/1336047/211695106-034c734c-86f2-4642-aa4d-0413e71af980.png">

Of course, both forms are legal, and it doesn't affect the mergability of this PR.

Reply:
@totten - yes I generally figure that when it comes to preferences I'd rather go with what has been generally agreed elsewhere rather than come up with our our negotiated version of what is more readable.  (My own readability preference in this case would be to concatenate it onto the end :-)).

Reply:
- as an aside - I've seen several things that phpstorm encouraged in the main inspections & the [Extended ones](https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-)  become required as we've gone up the php versions

Reply:
- on the readability I look at that line & think - WTF is 'cType'  so I don't get to the quoted part before my brain refuses to parse it so that's an 'unscannable' line from my POV

Reply:
And yes - I was right - phpstorm IS signalling an upcoming deprecation https://php.watch/versions/8.2/$%7Bvar%7D-string-interpolation-deprecated

Reply:
Since I wound up looking at this I decided to fix the variable names & then I wound up extracting the determination of the value - as is usually the case in civi those curly braces in a string point to a bit of code that is just nasty

Reply:
> on the readability I look at that line & think - WTF is 'cType' so I don't get to the quoted part before my brain refuses to parse it so that's an 'unscannable' line from my POV

Heh. Yeah. `cType` probably felt natural to whoever wrote it (at the time). Of course, if I look at that out-of-context, then I think: "oh, C language type -- `void*` or `char**`! Wait, why are we using C types in PHP?" 🙃

> ... I wound up extracting the determination of the value ...

If you're looking at some code and find some change which (subjectively) seems easier to read/understand, then power to you. 🚀 

> And yes - I was right - phpstorm IS signalling an upcoming deprecation https://php.watch/versions/8.2/$%7Bvar%7D-string-interpolation-deprecated

Just noting that this was incorrect, and it seems sort of material to topic.  Both `php.watch` and the [original RFC](https://wiki.php.net/rfc/deprecate_dollar_brace_string_interpolation) say that `{$var}` is preferred over `${var}` (since the latter is duplicative and less general). 

But the language benefits from having some flavor of `{}` , because bare `$var` can be ambiguous. I haven't seen any authority say `{$var}` is going away.

> I'd rather go with what has been generally agreed elsewhere rather than come up with our our negotiated version of what is more readable

Let's take that and treat it as a question. _What is the generally agreed approach from elsewhere vis-a-vis standards for variables-in-strings?_ Is it:

1. `"foo_$bar"`
2. `"foo_{$bar}"`
3. `'foo_' . $bar`
4. Other/None of the above

For "elsewhere", we can look to large communities like `drupal.org`, `wordpress.org`, `php-fig.org`, and `symfony.com`.  They all use PHP variables. They all have coding standards.  But _none of them_ offers an
answer to this question.  (There address some narrower cases -- like `t()` for Drupal prose and `sprintf()` for Symfony error-messages -- but this isn't prose. There's nothing that applies to this kind of string.)

Why not? Are Drupal developers too shy to haggle over obscure details? Is this a newfangled problem that coding-scientists have never seen before?  No... I submit that the absence points to answer (4). __They don't standardize this because it's not worth doing__. (*The cost of standardizing exceeds the benefit.*)

It seems to me that PHPStorm is the outlier, and... even so, PHPStorm is rather non-committal (*since it puts the "Disable" option right there in the popup...*).

(*To be clear -- I'm not saying that we should ignore everything PHPStorm says. I'm saying we should ignore it when it's wrong. 😈*)

Reply:
@totten I think the 'disable option right there in the popup' is a bit of a red herring - that option is there for all inspections.

But since you seem to care about this one I'll disable it. I go with phpstorm because it is a standard, picks up lots of good stuff - some of which have since been enforced, and helps us avoid bike shedding the details. But in terms of readability I think using actual words is much more important in this code than how the string is concatenated